[prave1n] iP

prave1n opened the pull request:
# MonoBot

> "Your mind is for having ideas, not holding them." – David Allen

MonoBot helps you keep track of your tasks so that you can focus on more important things. It's:

- text-based
- easy to learn
- ~~FAST~~ **SUPER FAST** to use

All you need to do is:

1. Download it from [here](https://github.com/nus-cs2103-AY2425S1/ip).
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you! 😉 

And it is **FREE**!

### Features:
- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

prave1n committed:
Level 0 (Rename, Greet, Exit)

prave1n committed:
Add echo

prave1n committed:
Add list to store tasks

prave1n committed:
Add ability to mark tasks

prave1n committed:
Add support for tracking three types of tasks

prave1n committed:
Test using the I/O redirection technique

prave1n committed:
Use Exceptions to handle errors

prave1n committed:
Add support for deleting tasks from the list

prave1n committed:
Test new branch

prave1n committed:
Save tasks in and load tasks from hard disk

prave1n committed:
Merge branch 'branch-Level-7'

prave1n committed:
Use LocalDateTime for time-sensitive tasks

prave1n committed:
Merge branch 'branch-Level-8'

prave1n committed:
Add Parser.java, TaskList.java, Ui.java

prave1n committed:
Add support for individual command classes

prave1n committed:
Divide classes into packages

prave1n committed:
Merge commit '81a9c537afe8492675eb253968939364d65db091'

prave1n committed:
Set up Gradle to build and run MonoBot

prave1n committed:
Add StorageTest.java and TaskListTest.java JUnit tests

prave1n committed:
Package app as JAR file

prave1n committed:
Add support to find task by keyword

prave1n committed:
Add JavaDoc comments

prave1n committed:
Edit code to comply with coding standards

prave1n committed:
Merge branch 'branch-A-JavaDoc'

prave1n committed:
Merge commit 'a39b856548a322ff7303215914af8905ed1e30cc'

prave1n committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/monobot/util/Parser.java
#	src/main/java/monobot/util/Ui.java

SeanFoongjt commented on src/main/java/monobot/command/InvalidCommand.java:
```diff
@@ -0,0 +1,31 @@
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+        ui.printError("Invalid monobot.command.Command. Valid commands are: \n" +
+                "list, todo, deadline, event, mark, unmark, bye");
```
Should the operator (+) be on the second line?

SeanFoongjt commented on src/test/java/monobot/util/StorageTest.java:
```diff
@@ -0,0 +1,95 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import static org.junit.jupiter.api.Assertions.*;
```
Should the imported Assertion classes be listed explicitly instead of with a wildcard?

SeanFoongjt commented on src/main/java/monobot/util/Parser.java:
```diff
@@ -0,0 +1,177 @@
+        case FIND:
+            return new FindCommand(getSearch(input));
+        default:
+            throw new IllegalArgumentException();
```
I like how you organised all the cases in the switch statement!

SeanFoongjt commented on src/main/java/monobot/util/Parser.java:
```diff
@@ -0,0 +1,177 @@
+    private static Task parseDeadline(String details) throws MonoBotException {
+        String[] deadlineDetails = details.split("/by", 2);
+        if (deadlineDetails.length != 2 || deadlineDetails[1].trim().isEmpty()) {
+            throw new MonoBotException("Due date/time of task is missing. " +
+                    "Note that format for adding a DEADLINE task is \n" +
```
Should the operators (+) here be on the succeeding line respectively?

SeanFoongjt commented on src/main/java/monobot/exception/MonoBotException.java:
```diff
@@ -0,0 +1,14 @@
+package monobot.exception;
+
+public class MonoBotException extends Exception {
+    String message;
+    public MonoBotException(String message) {
```
Perhaps there could be a space between these two lines?

SeanFoongjt reviewed (commented):
LGTM! Mostly just the wildcard imports and the placement of operators on linebreaks that seem to be violating the coding standard, good job! (I really like how you organised the code)

prave1n committed:
Add a GUI to the chatbot

prave1n committed:
Add GUI to chatbot

prave1n committed:
Merge branch 'Level-10'

prave1n committed:
Fix coding style violations with checkStyle

prave1n committed:
Merge branch 'A-CheckStyle'

jtooya commented on src/main/java/monobot/util/Storage.java:
```diff
@@ -0,0 +1,171 @@
+     * @param tasks  ArrayList of tasks to be saved to the file.
+     * @throws MonoBotException If an IO error occurs while writing the file.
+     */
+    public void save(ArrayList<Task> tasks) throws MonoBotException {
```
Perhaps the name could be more specific like saveToFile()?

jtooya commented on src/main/java/monobot/util/TaskList.java:
```diff
@@ -0,0 +1,116 @@
+    }
+
+    /**
+     * Gets a task from the list.
```
I think its great that you're very thorough in all your javadocs comments! Although would it make the code unnecessarily lengthy to document getters/setters in your code as well?

jtooya commented on src/main/java/monobot/util/Storage.java:
```diff
@@ -0,0 +1,171 @@
+import monobot.task.Deadline;
+import monobot.task.Event;
+import monobot.task.Task;
+import monobot.task.Todo;
```
Very clean and orderly import statements throughout the code! Good job!

jtooya reviewed (commented):
LGTM! Just a few very minor nits, but generally your code is clean and really easy to read through!

prave1n committed:
Support varargs for find command

prave1n committed:
Merge branch 'A-Varargs'

prave1n committed:
Add assertions to Parser, TaskList, and Ui classes

- Insert assert statements in the Parser class to validate method arguments and assumptions for parseCommand, parseTask, and other methods.
- Introduce assertions in TaskList methods such as addTask, deleteTask, and filterTasks to ensure valid input and internal consistency.
- Add assertions to the Ui class to check conditions before displaying output, ensuring non-null task lists and valid task indices.

These assertions help in detecting unexpected input and improve the robustness of the application during runtime.

prave1n committed:
Improve code quality and modify code to meet coding standards

prave1n committed:
Use the Streams feature to filter tasks for find command instructions

prave1n committed:
Merge pull request #1 from prave1n/branch-A-Assertions

Add assertions to Parser, TaskList, and Ui classes

prave1n committed:
Merge branch 'master' into branch-A-Streams

prave1n committed:
Merge branch 'master' into branch-A-CodeQuality

prave1n committed:
Merge pull request #3 from prave1n/branch-A-Streams

Use the Streams feature to filter tasks for find command instructions

prave1n committed:
Merge branch 'master' of github.com:prave1n/ip

prave1n committed:
Merge branch 'master' into branch-A-CodeQuality

prave1n committed:
Merge pull request #2 from prave1n/branch-A-CodeQuality

Improve code quality

prave1n committed:
Support mark, unmark and delete commands on multiple tasks

prave1n committed:
Set up Continuous Integration (CI)

prave1n committed:
Update user guide

prave1n committed:
Update image in user guide

prave1n committed:
Update build.gradle
